packages/@uppy/provider-views/src/SearchFilterInput.tsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable react/require-default-props */
+import { h, Fragment } from 'preact'
+import { useEffect, useState, useCallback } from 'preact/hooks'
+import { nanoid } from 'nanoid/non-secure'
+
+type Props = {
+  search: (query: string) => void
+  searchOnInput?: boolean
+  searchTerm?: string | null
+  showButton?: boolean
+  inputLabel: string
+  clearSearchLabel?: string
+  buttonLabel?: string
+  // eslint-disable-next-line react/require-default-props
+  clearSearch?: () => void
+  inputClassName: string
+  buttonCSSClassName?: string
+}
+
+export default function SearchFilterInput(props: Props): JSX.Element {
+  const {
+    search,
+    searchOnInput,
+    searchTerm,
+    showButton,
+    inputLabel,
+    clearSearchLabel,
+    buttonLabel,
+    clearSearch,
+    inputClassName,
+    buttonCSSClassName,
+  } = props
+  const [searchText, setSearchText] = useState(searchTerm ?? '')
+  // const debouncedSearch = debounce((q) => search(q), 1000)
+
+  const validateAndSearch = useCallback(
+    (ev: Event) => {
+      ev.preventDefault()
+      search(searchText)
+    },
+    [search, searchText],
+  )
+
+  const handleInput = useCallback(
+    (ev: Event) => {
+      const inputValue = (ev.target as HTMLInputElement).value
+      setSearchText(inputValue)
+      if (searchOnInput) search(inputValue)
+    },
+    [setSearchText, searchOnInput, search],
+  )
+
+  const handleReset = () => {
+    setSearchText('')
+    if (clearSearch) clearSearch()
+  }
+
+  const [form] = useState(() => {
+    const formEl = document.createElement('form')
+    formEl.setAttribute('tabindex', '-1')
+    formEl.id = nanoid()
+    return formEl
+  })
+
+  useEffect(() => {
+    document.body.appendChild(form)
+    form.addEventListener('submit', validateAndSearch)
+    return () => {
+      form.removeEventListener('submit', validateAndSearch)
+      document.body.removeChild(form)
+    }
+  }, [form, validateAndSearch])
+
+  return (
+    <Fragment>
+      <input
+        className={`uppy-u-reset ${inputClassName}`}
```

Comment:
That change seems unnecessary
```suggestion
        class={`uppy-u-reset ${inputClassName}`}
```

Reply:
`class` is not allowed in React/Preact.

Reply:
It is

Reply:
All our HTML elements / Preact components use `className`, would be weird to leave one instance inconsistent?

Reply:
+1 on consistency